test/sql/filter/test_transitive_filters.test

```diff
@@ -270,6 +270,7 @@ INSERT INTO vals2 SELECT * FROM vals1
 
 query IIII
 SELECT * FROM vals1, vals2 WHERE i>9 AND j<=l AND k>=i AND l<11
+ORDER BY 2 DESC, 4 DESC
```

Comment:
Could we add some more targeted tests for the multi-predicate piecewise join? While multi-predicate joins do exist in the test suite I think there are not that many. I would like to see e.g.:

* A test involving many (~10) predicates, each filtering out a certain subset of the rows (e.g. `a<z and b<z and c<z and d<z and e<z and ...` where `z=1..100` in random order and `a=10`, `b=20`, `c=30`, etc)
* The same test but with NULLs mixed in as well
* Joins on `BETWEEN` and `NOT BETWEEN` (this should now use the piecewise merge join as well, right?)
* Some more tests with joins on `COMPARE_DISTINCT_FROM` and `COMPARE_NOT_DISTINCT_FROM`, perhaps also checking the plan? I am not sure if these are correctly transformed to piecewise merge joins currently.
* Multi-predicate merge joins with many matches (similar to `test/sql/join/inner/test_unequal_join_duplicates.test` but with a merge join with multiple predicates)

Reply:
I want to be a bit deliberate about multiple inequalities because IEJoin will hopefully handle two or more of them eventually. So the coverage tests I'm adding will only have one range comparison, but I will add more general tests that will not impact coverage.

Reply:
`NOT DISTINCT FROM` is not handled by PWMJ (it is an equality condition).

Reply:
I'm a bit leery of generating too many matches because it is an N^2 algorithm, but I'll interpret that to mean that each side has more than one chunk?